course/views.py

```diff
@@ -1,3 +1,58 @@
-# from django.shortcuts import render
+from django.shortcuts import render, redirect, reverse
+from course.models import Review, TeacherCourse
+from course.forms import ReviewForm
+from django.contrib import messages
+from django.views.generic import DeleteView, UpdateView, CreateView, View
+from django.contrib.auth.mixins import LoginRequiredMixin
 
-# Create your views here.
+
+class CreateReviewView(LoginRequiredMixin, CreateView):
+    def get(self, request, course_id):
+        form_view = ReviewForm()
+        return render(request, 'course/create_review.html', {'form': form_view, 'course_id': course_id})
+
+    def post(self, request, course_id):
+        form = ReviewForm(request.POST)
+        if form.is_valid():
+            course = TeacherCourse.objects.filter(course_id=course_id)[0]
+            rating = request.POST.get('rating')
+            content = request.POST.get('content')
+            review = Review(student=request.user,  course=course, rating=rating, content=content)
+            review.save()
+            return redirect(reverse('reviews', args=[course_id]))
+        else:
+            messages.error(request, "You must enter rating")
+            return render(request, 'course/create_review.html', {'form': form, 'course_id': course_id})
+
+
+class ShowReviews(LoginRequiredMixin, View):
+    def get(self, request, course_id):
+        reviews = Review.objects.get_reviews_by_course(course_id)
+        if reviews:
+            return render(request, "course/reviews_by_course.html", {'reviews_by_course': reviews})
+        else:
+            return render(request, "course/reviews_by_course.html", {'reviews_by_course': []})
```

Comment:
same

Reply:
I have tested we can see the existed reviews, I will add a test for when there are no reviews existed.

Reply:
Done